pkg/portal/cluster/machines.go

```diff
@@ -59,6 +68,55 @@ func (c *client) Machines(ctx context.Context) (*MachineListInformation, error)
 	return c.fetcher.Machines(ctx)
 }
 
+// VM Allocation Status Stuff
+type VMAllocationStatus map[string]string
+
+func (c *client) VMAllocationStatus(ctx context.Context) (VMAllocationStatus, error) {
+	return c.fetcher.VMAllocationStatus(ctx)
+}
+
+func (f *realFetcher) VMAllocationStatus(ctx context.Context) (VMAllocationStatus, error) {
+	env := f.azureSideFetcher.env
+	subscriptionDoc := f.azureSideFetcher.subscriptionDoc
+	clusterRGName := f.azureSideFetcher.resourceGroupName
+	fpAuth, err := env.FPAuthorizer(subscriptionDoc.Subscription.Properties.TenantID, env.Environment().ResourceManagerEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	// Getting Virtual Machine resources through the Cluster's Resource Group
+	computeResources, err := newResourceClientFunction(env.Environment(), subscriptionDoc.ID, fpAuth).ListByResourceGroup(ctx, clusterRGName, "resourceType eq 'Microsoft.Compute/virtualMachines'", "", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	vmAllocationStatus := make(VMAllocationStatus)
+	virtualMachineClient := newVirtualMachineClientFunction(env.Environment(), subscriptionDoc.ID, fpAuth)
+	for _, res := range computeResources {
+		var vmName, allocationStatus string
+		vm, err := virtualMachineClient.Get(ctx, clusterRGName, *res.Name, mgmtcompute.InstanceView)
+		if err != nil {
+			f.log.Warn(err) // can happen when the ARM cache is lagging
+			// armResources = append(armResources, arm.Resource{
+			// 	Resource: res,
+			// })
+			continue
+		}
+
+		vmName = *vm.Name
+		instanceViewStatuses := vm.InstanceView.Statuses
+		for _, status := range *instanceViewStatuses {
+			if strings.Contains(*status.Code, "PowerState") {
```

Comment:
maybe doesn't matter too much, but isn't the real condition `string.HasPrefix(*status.Code, "PowerState/")` ?

Reply:
this condition can also be used, however, I have seen the statuses, they have powerstate only for actual power status of the VM.

Reply:
if the convention changes .. we will have to change the code anyway ..

Reply:
in any case, one is more correct than the other so I think we should go for the more precise one

Reply:
Okay .. I can do that ..

Reply:
done, resolving this conversation